zerver/tests/test_github.py

```diff
@@ -0,0 +1,103 @@
+            self.assertEqual(get_latest_github_release_version_for_repo("zulip-desktop"), "5.4.3")
+            m.assert_called_once_with("https://api.github.com/repos/zulip/zulip-desktop/releases/latest")
+
+        response_text = json.dumps({"tag_name": "5.4.4"})
+        with mock.patch("zerver.lib.github.requests.get", return_value=MockPythonResponse(response_text, 200)) as m:
+            # Versions starting without v
+            self.assertEqual(get_latest_github_release_version_for_repo("zulip-mobile"), "5.4.4")
+            m.assert_called_once_with("https://api.github.com/repos/zulip/zulip-mobile/releases/latest")
+            m.reset_mock()
+
+        with mock.patch("zerver.lib.github.requests.get", side_effect=Exception) as m:
+            self.assertEqual(get_latest_github_release_version_for_repo("zulip"), "")
+            m.reset_mock()
+
+    def test_verify_release_download_link(self) -> None:
+        headers = {"Location": "https://github-production-release-asset-2e65be.s3.amazonaws.com/60656871/68cdcf00-f2a7-11ea-897c-9b155c755721"}
+        with mock.patch("zerver.lib.github.requests.head", return_value=MockPythonResponse("", 302, headers=headers)) as m:
+            assert(verify_release_download_link("https://github.com/zulip/zulip-desktop/releases/download/v5.4.3/Zulip-Web-Setup-5.4.3.exe"))
+            m.assert_called_once_with("https://github.com/zulip/zulip-desktop/releases/download/v5.4.3/Zulip-Web-Setup-5.4.3.exe")
+            m.reset_mock()
+
+        with mock.patch("zerver.lib.github.requests.head", return_value=MockPythonResponse("", 404)) as m1:
+            with mock.patch("zerver.lib.github.logger.error") as m2:
+                assert(not verify_release_download_link("https://github.com/zulip/zulip-desktop/releases/download/v5.4.3/Zulip-Web-Setup-5.4.333.exe"))
+                m1.assert_called_once_with("https://github.com/zulip/zulip-desktop/releases/download/v5.4.3/Zulip-Web-Setup-5.4.333.exe")
+                m1.reset_mock()
+                m2.assert_called_once()
+                m2.reset_mock()
+
+        with mock.patch("zerver.lib.github.requests.head", side_effect=Exception) as m:
+            assert(not verify_release_download_link("https://github.com/zulip/zulip-desktop/releases/download/v5.4.3/Zulip-Web-Setup-5.4.333.exe"))
+            m.reset_mock()
+
+    def test_get_latest_github_release_download_link_for_platform(self) -> None:
+        with mock.patch("zerver.lib.github.get_latest_github_release_version_for_repo", return_value="5.4.3") as m1:
+            with mock.patch("zerver.lib.github.verify_release_download_link", return_value=True) as m2:
+                self.assertEqual(
+                    get_latest_github_release_download_link_for_platform("windows"),
+                    "https://github.com/zulip/zulip-desktop/releases/download/v5.4.3/Zulip-Web-Setup-5.4.3.exe"
+                )
+                m1.assert_called_once_with("zulip-desktop")
```

Comment:
If we’re going to verify that the call was not cached, we should use `cache_delete` to be sure that it wasn’t.